tests/integration-tests/tests/iam/test_iam.py

```diff
@@ -332,3 +336,134 @@ def test_s3_read_write_resource(region, pcluster_config_reader, s3_bucket_factor
     # Check S3 resources
     check_s3_read_resource(region, cluster, get_policy_resources(config, enable_write_access=False))
     check_s3_read_write_resource(region, cluster, get_policy_resources(config, enable_write_access=True))
+
+
+@pytest.mark.parametrize("iam_resource_prefix", ["name-prefix-", "/path-prefix/", "/path-prefix/name-prefix-"])
+@pytest.mark.usefixtures("os", "instance")
+def test_iam_resource_prefix(
+    register_resource_prefix_cli_credentials,
+    region,
+    pcluster_config_reader,
+    clusters_factory,
+    test_datadir,
+    scheduler_commands_factory,
+    s3_bucket_factory,
+    s3_bucket,
+    iam_resource_prefix,
+):
+    bucket_name = s3_bucket
+    cfn_client, _, iam_client, _ = _create_boto3_clients(region)
+
+    create_config, update_config = _get_config_create_and_update(test_datadir)
```

Comment:
The `update_config` variable is never used. 
Is it because the PR is actually a draft and we expect to use it later on?
If this is the case, it would be better to mark the PR as a draft